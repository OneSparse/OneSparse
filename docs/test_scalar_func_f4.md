# Scalar `float4` Functions

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
``` postgres-console
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
```
Test construction of min, zero and max values:
``` postgres-console
select '-1'::float4::scalar;
    scalar    
--------------
 f4:-1.000000
(1 row)

select '0'::float4::scalar;
   scalar    
-------------
 f4:0.000000
(1 row)

select '1'::float4::scalar;
   scalar    
-------------
 f4:1.000000
(1 row)

```
Test setting a scalar value from max to 2
``` postgres-console
select set('1'::float4::scalar, 2);
     set     
-------------
 f4:2.000000
(1 row)

```
Test various casting functions used by the CREATE CAST machinery:
``` postgres-console
select scalar_float4((-1)::float4);
 scalar_float4 
---------------
 f4:-1.000000
(1 row)

select scalar_float4((0)::float4);
 scalar_float4 
---------------
 f4:0.000000
(1 row)

select scalar_float4((1)::float4);
 scalar_float4 
---------------
 f4:1.000000
(1 row)

select float4_scalar((-1)::float4::scalar);
 float4_scalar 
---------------
            -1
(1 row)

select float4_scalar((0)::float4::scalar);
 float4_scalar 
---------------
             0
(1 row)

select float4_scalar((1)::float4::scalar);
 float4_scalar 
---------------
             1
(1 row)

```