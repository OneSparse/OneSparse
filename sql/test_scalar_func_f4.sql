set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select '-1'::float4::scalar;
select '0'::float4::scalar;
select '1'::float4::scalar;
