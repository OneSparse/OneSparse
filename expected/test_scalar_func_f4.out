set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select '-1'::float4::scalar;
    scalar    
--------------
 f4:-1.000000
(1 row)

select '0'::float4::scalar;
   scalar    
-------------
 f4:0.000000
(1 row)

select '1'::float4::scalar;
   scalar    
-------------
 f4:1.000000
(1 row)

select set('1'::float4::scalar, 2);
     set     
-------------
 f4:2.000000
(1 row)

select scalar_float4((-1)::float4);
 scalar_float4 
---------------
 f4:-1.000000
(1 row)

select scalar_float4((0)::float4);
 scalar_float4 
---------------
 f4:0.000000
(1 row)

select scalar_float4((1)::float4);
 scalar_float4 
---------------
 f4:1.000000
(1 row)

select float4_scalar((-1)::float4::scalar);
 float4_scalar 
---------------
            -1
(1 row)

select float4_scalar((0)::float4::scalar);
 float4_scalar 
---------------
             0
(1 row)

select float4_scalar((1)::float4::scalar);
 float4_scalar 
---------------
             1
(1 row)

